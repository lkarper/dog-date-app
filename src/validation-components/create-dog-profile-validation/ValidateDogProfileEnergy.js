import React, { useEffect } from 'react';

const ValidateDogProfileEnergy = (props) => {

    const { energy, energyError, setEnergyError } = props;

    useEffect(() => {
        if (energy) {
            setEnergyError('');
        } else {
            setEnergyError(`You must select your dog's energy level.`);
        }
    }, [energy, setEnergyError]);

    if (energy) {
        return <p id='energy-validator'>Energy set to: '{energy}'</p>;
    }

    return <p id='energy-validator'>{energyError}</p>;
}

export default ValidateDogProfileEnergy;
