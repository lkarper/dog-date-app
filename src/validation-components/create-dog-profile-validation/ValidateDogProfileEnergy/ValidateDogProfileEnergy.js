import React, { useEffect } from 'react';

const ValidateDogProfileEnergy = (props) => {

    const { energy, energyError, setEnergyError } = props;

    useEffect(() => {
        if (energy) {
            setEnergyError('');
        } else {
            setEnergyError(`You must select your dog's energy level.`);
        }
    }, [energy, setEnergyError]);

    if (energy) {
        return (
            <p 
                className='ValidateDogProfileEnergy__validator valid'
                id='energy-validator'
            >
                Energy set to: '{energy}'
            </p>
        );
    }

    return (
        <p 
            className='ValidateDogProfileEnergy__validator error'            
            id='energy-validator'
        >
            {energyError}
        </p>
    );
}

export default ValidateDogProfileEnergy;
