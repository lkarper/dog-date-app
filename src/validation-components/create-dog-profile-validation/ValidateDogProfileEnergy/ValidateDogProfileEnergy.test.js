import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ValidateDogProfileEnergy from './ValidateDogProfileEnergy';

describe('ValidateDogProfileEnergy component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<ValidateDogProfileEnergy />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<ValidateDogProfileEnergy />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
