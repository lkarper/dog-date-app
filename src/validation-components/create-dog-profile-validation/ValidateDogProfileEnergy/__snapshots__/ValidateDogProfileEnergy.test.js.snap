// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidateDogProfileEnergy component renders the UI as expected 1`] = `
<ValidateDogProfileEnergy
  energy=""
  energyError=""
  setEnergyError={[Function]}
>
  <p
    className="ValidateDogProfileEnergy__validator error"
    id="energy-validator"
  />
</ValidateDogProfileEnergy>
`;
